src/Illuminate/Database/Eloquent/Factories/Factory.php

```diff
@@ -802,16 +804,16 @@ public function withoutParents($parents = [])
      */
     public function getConnectionName()
     {
-        return $this->connection;
+        return enum_value($this->connection);
     }
 
     /**
      * Specify the database connection that should be used to generate models.
      *
-     * @param  string  $connection
+     * @param  \UnitEnum|string  $connection
      * @return static
      */
-    public function connection(string $connection)
```

Comment:
Technically this is backwards incompatible change for classes extending this class and overriding this method with a typed signature, as they would have to update theirs. I'm not sure if such cases are considered to be breaking changes; if so I can port just this change to `master` if preferred.